src/algebra/module/linear_map.lean

```diff
@@ -118,6 +118,12 @@ variables (f g)
 
 @[simp] lemma map_smul (c : R) (x : M) : f (c • x) = c • f x := f.map_smul' c x
 
+@[simp, priority 900]
+lemma map_smul_of_tower {S : Type*} [has_scalar S R] [has_scalar S M]
```

Comment:
Could you please flip the role of `S` and `R`? Typically `R` is a base ring, and `S` is an `R`-algebra.